### PR TITLE
Rework baselines

### PR DESCRIPTION
**Pull request by alafage**
Current baseline implementations carry lots of repetitions. To simplify them this pull request proposes the following changes:

1. Classification Single/Ensemble routines are no more abstract. They can be used directly with models, losses, etc.
2. Unique baseline for each backbone (ResNet, WideResNet, etc.) where the version (ex: vanilla, packed, etc.) is passed as an argument.

These changes are expected to:

1. reduce considerably the number of files for the baselines
2. ease the use of routines to train custom models

- [x] Routines are not more abstract
- [x] ResNet baseline
- [x] WideResNet baseline
- [x] ResNet experiments
- [x] WideResNet experiments
- [x] Custom optimizer management for BatchEnsemble
- [x] Simplify the arguments (of packed, etc) but also of the routine (metrics)
- [x] Simplify the argument consistency tests
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by o-laurent** [comment]
You also moved loading weights inside the model #11

File: torch_uncertainty/optimization_procedures.py [hunk 1]
```diff
@@ -176,17 +179,73 @@ def optim_imagenet_resnet50_A3(
...
+    elif arch_name == "resnet50":
+        if ds_name == "cifar10":
+            procedure = optim_cifar10_resnet50
+        elif ds_name == "cifar100":
+            procedure = optim_cifar100_resnet50
+    elif arch_name == "wideresnet28x10":
+        if ds_name == "cifar10" or ds_name == "cifar100":
+            procedure = optim_cifar10_wideresnet
+
+    if model_name == "batch_ensemble":
```
Comment (alafage) [inline torch_uncertainty/optimization_procedures.py:248]: I think it should be:
```py
if model_name == "batched":
    ...
```

Comment (o-laurent) [reply torch_uncertainty/optimization_procedures.py:248]: You're right, however, I was wondering whether batched was the right name.

Comment (o-laurent) [reply torch_uncertainty/optimization_procedures.py]: I fixed the optimizer, but we may need to look at the names we gave to the different methods.

**Review by o-laurent — COMMENTED** [review]
Starting to look nice!

**Pull request merged**